fe/fe-core/src/main/java/org/apache/doris/nereids/util/TypeCoercionUtils.java

```diff
@@ -1831,4 +1834,243 @@ private static boolean supportCompare(DataType dataType) {
         }
         return true;
     }
+
+    public static void validateDataType(DataType dataType) {
```

Comment:
should not add this function in TypeCoercionUtils. add this function in DataType and Type directly